### PR TITLE
verify magic for all frames

### PR DESCRIPTION
**Pull request by arnej27959**
I'm writing an implementation of ZstdInputStream for contribution, and noticed a minor bug.
When decompressing multiple frames all calls to verifyMagic would use the address of the first frame.

### CODE REVIEW

**Comment by martint** [comment]
Thanks! Can you add a test for this? We can add a file with multiple frames an a bad magic header in one of the frames other than the first one.

**Comment by arnej27959** [comment]
ok, I duplicated src/test/resources/data/zstd/multiple-frames.zst and changed the magic in the second frame.
this triggers as expected if I revert the first commit:
`Method TestZstd.testVerifyMagicInAllFrames()[pri:0, instance:io.airlift.compress.zstd.TestZstd@5e600dd5] should have thrown an exception of type class io.airlift.compress.MalformedInputException`

File: src/test/java/io/airlift/compress/zstd/TestZstd.java [hunk 1]
```diff
@@ -182,4 +182,16 @@ public void testGetDecompressedSize(DataSet dataSet)
...
+    @Test(expectedExceptions = MalformedInputException.class, expectedExceptionsMessageRegExp = "Invalid magic prefix.*")
+    public void testVerifyMagicInAllFrames()
+            throws IOException
+    {
+        Compressor compressor = getCompressor();
+        byte[] compressed = Resources.toByteArray(getClass().getClassLoader().getResource("data/zstd/bad-second-frame.zst"));
+        byte[] uncompressed = Resources.toByteArray(getClass().getClassLoader().getResource("data/zstd/multiple-frames"));
+        byte[] output = new byte[uncompressed.length];
+        getDecompressor().decompress(compressed, 0, compressed.length, output, 0, output.length);
+        assertByteArraysEqual(uncompressed, 0, uncompressed.length, output, 0, output.length);
```
Comment (martint) [inline src/test/java/io/airlift/compress/zstd/TestZstd.java:195]: This is unnecessary, as the test is expected to fail.

Comment (arnej27959) [reply src/test/java/io/airlift/compress/zstd/TestZstd.java]: fixed. do you want me to squish my 3 commits into 1?

File: src/test/java/io/airlift/compress/zstd/TestZstd.java [hunk 2]
```diff
@@ -182,4 +182,15 @@ public void testGetDecompressedSize(DataSet dataSet)
         System.arraycopy(compressed, 0, compressedWithPadding, padding, compressedLength);
         assertEquals(ZstdDecompressor.getDecompressedSize(compressedWithPadding, padding, compressedLength), originalUncompressed.length);
     }
+
+    @Test(expectedExceptions = MalformedInputException.class, expectedExceptionsMessageRegExp = "Invalid magic prefix.*")
```
[the remark this answers is not in the document]
Comment (arnej27959) [reply src/test/java/io/airlift/compress/zstd/TestZstd.java]: > use org.assertj.core.api.Assertions.assertThatThrownBy instead

done.

**Review by findepi — APPROVED** [review]
Can you please squash commits?

**Comment by arnej27959** [comment]
commits squashed into one now.

**Comment by arnej27959** [comment]
Note: Githubs tells me
```
1 workflow awaiting approval
First-time contributors need a maintainer to approve running workflows
```
and therefore the CI job has not run for this pull request.

File: src/test/java/io/airlift/compress/zstd/TestZstd.java [hunk 3]
```diff
@@ -85,14 +86,16 @@ public void testConcatenatedFrames()
...
     public void testInvalidSequenceOffset()
             throws IOException
     {
         byte[] compressed = Resources.toByteArray(getClass().getClassLoader().getResource("data/zstd/offset-before-start.zst"));
         byte[] output = new byte[compressed.length * 10];
 
-        getDecompressor().decompress(compressed, 0, compressed.length, output, 0, output.length);
+        assertThatThrownBy(() -> getDecompressor().decompress(compressed, 0, compressed.length, output, 0, output.length))
+                .isInstanceOf(MalformedInputException.class)
+                .hasMessageStartingWith("Input is corrupted: offset=894");
```
Comment (martint) [inline src/test/java/io/airlift/compress/zstd/TestZstd.java:98]: Thanks for improving this test. However, since this change is unrelated to what this commit is fixing, please move it to a separate commit to keep changes self-contained.

**Review by martint — COMMENTED** [review]
One last minor comment: please capitalize the commit message.

File: src/test/java/io/airlift/compress/zstd/TestZstd.java (hunk 3 above)
Comment (arnej27959) [reply src/test/java/io/airlift/compress/zstd/TestZstd.java:98]: split into separate commit now.
